File: python/PolyHash.py
import hashlib

"""
Calculate hashes of strings
"""

def md5(data):
    """Calculate the md5 hash of a string
    Args:
        data (str): The string to digest

    Returns:
        string: The digest in hexadecimal
    """
    return hashlib.md5(data).hexdigest()

def sha1(data):
    """Calculate the sha1 hash of a string
    Args:
        data (str): The string to digest

    Returns:
        string: The digest in hexadecimal
    """
    return hashlib.sha1(data).hexdigest()

def sha256(data):
    """Calculate the sha256 hash of a string
    Args:
        data (str): The string to digest

    Returns:
        string: The digest in hexadecimal
    """
    return hashlib.sha256(data).hexdigest()

def sha512(data):
    """Calculate the sha512 hash of a string
    Args:
        data (str): The string to digest

    Returns:
        string: The digest in hexadecimal
    """
    return hashlib.sha512(data).hexdigest()

File: node/src/PolyHash.js
import forge from 'node-forge';

/**
 * Calculate hashes of strings
 */
export const PolyHash = {
	/**
	 * Calculate the md5 hash of a string
	 * @param {String} data  The string to digest
	 * @return {String} The digest in hexadecimal
	 */
	md5(data) {
		return PolyHash._hash('md5', data);
	},

	/**
	 * Calculate the sha1 hash of a string
	 * @param {String} data  The string to digest
	 * @return {String} The digest in hexadecimal
	 */
	sha1(data) {
		return PolyHash._hash('sha1', data);
	},

	/**
	 * Calculate the sha256 hash of a string
	 * @param {String} data  The string to digest
	 * @return {String} The digest in hexadecimal
	 */
	sha256(data) {
		return PolyHash._hash('sha256', data);
	},

	/**
	 * Calculate the sha512 hash of a string
	 * @param {String} data  The string to digest
	 * @return {String} The digest in hexadecimal
	 */
	sha512(data) {
		return PolyHash._hash('sha512', data);
	},

	/**
	 * Private function to calculate hashes for the given algorithm
	 * @param {String} algo  An algorithm on the forge.md namespace
	 * @param {String} data  The string to digest
	 * @return {String} The digest in hexadecimal
	 * @private
	 */
	_hash(algo, data) {
		const md = forge.md[algo].create();
		md.update(data);
		return md.digest().toHex();
	},
};
